core/src/test/java/org/apache/iceberg/rest/TestRESTCatalog.java

```diff
@@ -152,6 +164,30 @@ public void createCatalog() throws Exception {
     adapterForRESTServer =
         Mockito.spy(
             new RESTCatalogAdapter(backendCatalog) {
+              @Override
+              public <T extends RESTResponse> T handleRequest(
```

Comment:
instead of handling this globally for all tests and introducing `idempHeaderExpectation` and `advertiseIdempInConfig` it's probably better to use a custom `RESTCatalogAdapter` for the 2 new tests that you're adding. One example where this is done is in `verifyNamespaceExistsFallbackToGETRequest` and we could use the same approach in the new tests as well

Reply:
Sounds good! Fixed!